vhost-device-console/src/vhu_console.rs

```diff
@@ -165,28 +168,53 @@ impl VhostUserConsoleBackend {
             output_queue: Queue::new(),
             stdin: None,
             stream: None,
-            listener: None,
+            tcp_listener: None,
+            uds_listener: None,
             rx_event: EventFd::new(EFD_NONBLOCK).map_err(|_| Error::EventFdFailed)?,
             rx_ctrl_event: EventFd::new(EFD_NONBLOCK).map_err(|_| Error::EventFdFailed)?,
             exit_event: EventFd::new(EFD_NONBLOCK).map_err(|_| Error::EventFdFailed)?,
             mem: None,
         })
     }
 
-    pub fn assign_input_method(&mut self, tcpaddr_str: String) -> Result<()> {
-        if self.controller.read().unwrap().backend == BackendType::Nested {
-            // Enable raw mode for local terminal if backend is nested
-            enable_raw_mode().expect("Raw mode error");
+    pub fn assign_input_method(&mut self, vm_sock_addr: String) -> Result<()> {
+        let backend_type = self.controller.read().unwrap().backend;
+        match backend_type {
+            BackendType::Nested => {
+                // Enable raw mode for local terminal if backend is nested
+                enable_raw_mode().expect("Raw mode error");
+                let stdin_fd = io::stdin().as_raw_fd();
+                let stdin: Box<dyn Read + Send + Sync> = Box::new(io::stdin());
+                self.stdin = Some(stdin);
+                Self::epoll_register(self.epoll_fd.as_raw_fd(), stdin_fd, epoll::Events::EPOLLIN)
+                    .map_err(|_| Error::EpollAdd)?;
+            }
 
-            let stdin_fd = io::stdin().as_raw_fd();
-            let stdin: Box<dyn Read + Send + Sync> = Box::new(io::stdin());
-            self.stdin = Some(stdin);
+            BackendType::Network => {
+                let tcp_listener = TcpListener::bind(&vm_sock_addr).map_err(|e| {
+                    eprintln!("Failed to bind TCP listener to {}: {}", vm_sock_addr, e);
+                    Error::CreateBackendFailed
+                })?;
+                self.tcp_listener = Some(tcp_listener);
+            }
 
-            Self::epoll_register(self.epoll_fd.as_raw_fd(), stdin_fd, epoll::Events::EPOLLIN)
-                .map_err(|_| Error::EpollAdd)?;
-        } else {
-            let listener = TcpListener::bind(tcpaddr_str).expect("asdasd");
```

Comment:
wow, we had `asdasd` ?!?

thanks for fixing it, but please do in a separate patch